src/ServiceProvider.php

```diff
@@ -21,6 +25,27 @@ public function boot()
         Queue::looping(function () {
             app('log')->reset();
         });
+
+        /**
+         * Add the trace ID to context if its not already present.
+         *
+         * Web requests will have this value set by the load balancer and so will add it to context.
+         * Queue jobs will have any context persisted when they are dispatched, so for jobs that are
+         * dispatched from a web request (ie not a cron) will already have this value hydrated and
+         * won't try to set another value. This will also work across fan-out jobs.
+         * Crons won't have this available ever.
+         *
+         * Any values added to context will also end up in our logs under the `extra` key. This is
+         * useful for following code across services for a single web request.
```

Comment:
Block: none of this is true within the context of this library.

Reply:
This is a public library so there is no guarantee that others have this exact setup

Reply:
Hmm okay. What are you suggesting instead then? This is an open-source library, but it does contain our common configuration as a means to deduplicate it from our services. If we are worried about others not having this setup, then it probably shouldn't exist

Reply:
Delete all of the comments that are unrelated to this library